File: Decorator/PHP/Espresso.php
<?php
  class Espresso extends Beverage {
    public function __construct() {
      parent::__construct();
      $this->description = " Espresso";
    }

    public function cost() {
      return 5.6;
    }

    public function getDescription() {
      return $this->description;
    }
  }

File: Decorator/Go/decorator/espresso.go
package decorator

type Espresso struct {
  *Coffee
}

func NewEspresso() *Espresso {
  return &Espresso{&Coffee{price: 1.0, description: "Espresso"}}
}

func (self *Espresso) Cost() float64 {
  return self.price
}

func (self Espresso) GetDescription() string {
  return self.description
}
